pkg/sql/show_fingerprints.go

```diff
@@ -176,41 +213,54 @@ func (n *showFingerprintsNode) nextTenant(params runParams) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+	n.run.values[0] = tree.NewDString(string(tinfo.Name))
 
 	tid, err := roachpb.MakeTenantID(tinfo.ID)
 	if err != nil {
 		return false, err
 	}
 
+	// We want to write a protected timestamp record at the earliest timestamp
+	// that the fingerprint query is going to read from. When fingerprinting
+	// revisions, this will be the specified start time.
+	tsToProtect := params.p.EvalContext().Txn.ReadTimestamp()
+	if n.options != nil && !n.options.startTimestamp.IsEmpty() {
+		tsToProtect = n.options.startTimestamp
+	}
```

Comment:
Should we verify that start_time < ReadTimestamp()?

Reply:
yup, i added a check to return an assertion failed.